// src/jsxte-render-error.ts
var mapReverse = (arr, fn) => {
  const result = [];
  for (let i = arr.length - 1; i >= 0; i--) {
    result.push(fn(arr[i]));
  }
  return result;
};
var JsxteRenderError = class _JsxteRenderError extends Error {
  constructor(message, insideTag, causedBy) {
    super(message, { cause: causedBy });
    this.baseMessage = "";
    this.parentTags = [];
    this.name = "JsxteRenderError";
    this.baseMessage = message;
    if (insideTag) {
      this.parentTags.push(insideTag);
    }
    if (this.cause == null) {
      Object.defineProperty(this, "cause", {
        value: causedBy,
        enumerable: true,
        writable: true
      });
    }
  }
  static is(err) {
    return err instanceof _JsxteRenderError;
  }
  /**
   * @internal
   */
  pushParent(tag) {
    this.parentTags.push(tag);
  }
  /**
   * @internal
   */
  regenerateMessage() {
    this.message = `The below error has occurred in:
${mapReverse(
      this.parentTags.filter((t) => t !== ""),
      (tag) => `<${tag}>`
    ).join("\n")}

${this.baseMessage}`;
  }
};
export {
  JsxteRenderError
};
